lib/fluent/plugin/elasticsearch_error_handler.rb

```diff
@@ -85,11 +85,17 @@ def handle_error(response)
     errors.each_key do |key|
       case key
       when 'out_of_memory_error'
-        raise ElasticsearchOutOfMemory, "Elasticsearch has exhausted its heap, retrying"
+        error_message = "Elasticsearch has exhausted its heap, retrying"
+        @plugin.log.error(error_message)
+        raise ElasticsearchOutOfMemory, error_message
       when 'es_rejected_execution_exception'
-        raise BulkIndexQueueFull, "Bulk index queue is full, retrying"
+        error_message = "Bulk index queue is full, retrying"
+        @plugin.log.error(error_message)
+        raise BulkIndexQueueFull, error_message
       else
-        raise ElasticsearchError, "Elasticsearch errors returned, retrying  #{response}"
+        error_message = "Elasticsearch errors returned, retrying  #{response}"
```

Comment:
Won't this format the string in all its glory even when we are not going to log it via `@plugin.log.debug()`?  Can we avoid that?

Reply:
It will, but fluentd does not have a  'debug?' method so we either have to not log at all or take the pain.  Fluent will only log if its debug but we would get the string expansion

Reply:
I stand corrected, they have an on_debug method that takes a block.